codec-http/src/main/java/io/netty/handler/codec/http/DefaultLastHttpContent.java

```diff
@@ -107,32 +108,24 @@ private void appendHeaders(StringBuilder buf) {
     }
 
     private static final class TrailingHttpHeaders extends DefaultHttpHeaders {
-        private static final class TrailingHttpHeadersNameConverter extends HttpHeadersNameConverter {
-            TrailingHttpHeadersNameConverter(boolean validate) {
-                super(validate);
-            }
+        private static final class TrailingHttpHeadersNameValidator implements
+                                                                    DefaultHeaders.NameValidator<CharSequence> {
+
+            public static final TrailingHttpHeadersNameValidator INSTANCE = new TrailingHttpHeadersNameValidator();
```

Comment:
> only public if used in other classes / outer class

+1. Set the accessibility as if it were not an inner class...unless you are trying to avoid synthetic method generation or other oddities.